src/seedu/addressbook/data/CommandHistory.java

```diff
@@ -0,0 +1,27 @@
+package seedu.addressbook.data;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+
+/**
+ * Class to save the history of commands executed.
+ */
+public class CommandHistory {
+    private ArrayList<String> commandHistory = new ArrayList<>();
+
+    public void addToHistory(String executedCommand) {
+        this.commandHistory.add(executedCommand);
+    }
+
+    public String toString() {
+        Iterator iter = this.commandHistory.iterator();
+        StringBuilder result = new StringBuilder();
+
+        while (iter.hasNext()) {
+            result.append(iter.next()+ "\n");
```

Comment:
Good that you used a string builder! However note that using string builder in this way reduces it efficiency because java uses a buffer underneath its hood when you concat a string.